### ISSUE TITLE
Using conditional operator causes VerifyError

### ISSUE THREAD

**Issue by Waldo1637** [Sable/soot#484]
In some cases, when the java ternary conditional operator appears in the input file, Soot produces code which does not pass JVM verification. The following Java method (when compiled and passed to Soot in .class form) causes this behavior.
```java
public boolean test(int m) {
    return m < 0 || m > 0 ? false : 5 < m;
}
```

This issue only arises in specific uses of the conditional statement. Other variations of this conditional statement do NOT cause the verify error. Also, using java source instead of class file as input to Soot does not cause this error. The following examples do not appear to cause VerifyError to be thrown by the JVM:
- using a more simple conditional statement
- using AND instead of OR in the conditional statement
- using true instead of false for the first value
- swapping the values

The test method (shown above) produces the following Jimple code:
```
      public boolean test(int) {
1         VerifyErrorTernary r0;
2         int i0;
3         boolean $z0;
4         r0 := @this;
5         i0 := @parameter0; 
6         if i0 < 0 goto label1;
7         if i0 <= 0 goto label2;
      label1:
8         $z0 = 0;
9         goto label3;
      label2:
10        if 5 >= i0 goto label3;
11        $z0 = 1;
12        goto label3;
      label3:
13        return $z0;
      }
```

The JVM throws the VerifyError because of the control flow path [1...7, 10,13]. Notice along that path, $z0 is never defined hence the VerifyError.

**Comment by ericbodden** [comment]
@Waldo1637 thanks a lot for reporting this. Could you please let us know which command line you are using for Soot to reproduce this issue? For us this is important to know because Soot comprises a variety of frontends and backends (some for source code, some for bytecode) and to drill down the problem we need to know which ones are being used.

If you had a complete test case in form of a .java or .class file that would be even better.

**Comment by Waldo1637** [comment]
I have placed a sample classfile [on Dropbox](https://www.dropbox.com/s/oimq51q2k58l0q8/VerifyErrorTernary.class?dl=0).
The command line I used to run Soot is `java -cp soot.jar soot.Main -pp -cp . VerifyErrorTernary`

**Comment by Alexandre-Bartel** [comment]
Thank you for the sample. This should have been fixed by c6d78d4430bd77d06ec1b37b8af8192006510b86.
